integration-cli/check_test.go

```diff
@@ -38,37 +42,64 @@ const (
 )
 
 var (
-	testEnv *environment.Execution
+	testEnvOnce sync.Once
+	testEnv     *environment.Execution
 
 	// the docker client binary to use
 	dockerBinary = ""
 
-	testEnvOnce sync.Once
+	baseContext context.Context
 )
 
-func init() {
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	os.Exit(testRun(m))
+}
+
+func testRun(m *testing.M) (ret int) {
+	// Global set up
+
 	var err error
 
-	testEnv, err = environment.New()
+	shutdown := testutil.ConfigureTracing()
+	ctx, span := otel.Tracer("").Start(context.Background(), "integration-cli/TestMain")
+	defer func() {
+		if err != nil {
+			span.SetStatus(codes.Error, err.Error())
+			ret = 255
+		} else {
+			if ret != 0 {
+				span.SetAttributes(attribute.Int("exitCode", ret))
+				span.SetStatus(codes.Error, "m.Run() exited with non-zero code")
+			}
+		}
+		span.End()
+		shutdown(ctx)
+	}()
+
+	baseContext = ctx
```

Comment:
I wish `testing.T` / `testing.M` would have the ability to attach a context (so that we didn't need to use a package-level var for this) 😢